api/src/main/java/io/jsonwebtoken/lang/Services.java

```diff
@@ -0,0 +1,51 @@
+package io.jsonwebtoken.lang;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+
+/**
+ * Helper class for loading services from the classpath, using a {@link ServiceLoader}. Decouples loading logic for
+ * better separation of concerns and testability.
+ */
+public final class Services {
+
+    /**
+     * Loads and instantiates all service implementation of the given SPI class and returns them as a List.
+     *
+     * @param spi The class of the Service Provider Interface
+     * @param <T> The type of the SPI
+     * @return An unmodifiable list with an instance of all available implementations of the SPI. No guarantee is given
+     * on the order of implementations, if more than one.
+     */
+    public static <T> List<T> loadAllAvailableImplementations(Class<T> spi) {
```

Comment:
This method name feels a little verbose to me.  How about just `loadAll` since this better parallels the `loadFirst` nomenclature below (i.e. it's not `loadFirstAvailableImplementation`, it's just a nice clean `loadFirst`).

Reply:
Also, is this method called anywhere else other than by `loadFirst` ?  If not, we should make it private to reduce our exposure area of things we need to support for users.